collector/collector.go

```diff
@@ -12,73 +12,81 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Collector runs a tool to collect the data needed to monitor a single
+// Package collector runs a tool to collect the data needed to monitor a single
 // Certificate Transparency Log, which can then be used to check that it is
 // adhering to RFC 6962.
-//
-// TODO(katjoyce): Once it contains more functionality, turn this into a package
-// so that it can be run on multiple Logs by spinning up one per Log.
-package main
+package collector
 
 import (
 	"context"
-	"flag"
+	"errors"
+	"fmt"
 	"net/http"
 	"sync"
 	"time"
 
-	"github.com/golang/glog"
 	ct "github.com/google/certificate-transparency-go"
 	"github.com/google/monologue/client"
 	"github.com/google/monologue/ctlog"
 	"github.com/google/monologue/rootsgetter"
 	"github.com/google/monologue/sthgetter"
-	"github.com/google/monologue/storage/print"
+	"github.com/google/monologue/storage"
 )
 
-var (
-	rootsGetPeriod = flag.Duration("roots_get_period", 5*time.Minute, "How regularly the monitor should get root certificates from the Log")
-	sthGetPeriod   = flag.Duration("sth_get_period", 1*time.Minute, "How regularly the monitor should get an STH from the Log")
-	logURL         = flag.String("log_url", "", "The URL of the Log to monitor, e.g. https://ct.googleapis.com/pilot/")
-	logName        = flag.String("log_name", "", "A short, snappy, canonical name for the Log to monitor, e.g. google_pilot")
-	b64PubKey      = flag.String("public_key", "", "The base64-encoded public key of the Log to monitor")
-	mmd            = flag.Duration("mmd", 24*time.Hour, "The Maximum Merge Delay for the Log")
-)
+// Config contains all of the configuration details for running the collector
+// for a particular Log.
+type Config struct {
+	// Details of the Log to collect data from.
+	Log *ctlog.Log
+	// How regularly the monitor should get an STH from the Log.
+	// To disable STH-getting, set to 0.
+	STHGetPeriod time.Duration
```

Comment:
nit: Putting the "Get" at the start of the name makes them match the API names more closely, which makes the names a bit more memorable.
```suggestion
	GetSTHPeriod time.Duration
```

Reply:
Done.